pkg/agent/agent_linux.go

```diff
@@ -166,30 +166,6 @@ func (i *Initializer) saveHostRoutes() error {
 	return nil
```

Comment:
@wenyingd : I saw there is a difference between `GetAllIPNetsByName` and `GetIPNetsByLink ` called by `GetInterfaceConfig` that the former ignores link local addresses. Should we always ignore link local addresses? Could/should we change `GetIPNetsByLink` to do that too?

Reply:
I saw GetInterfaceConfig is used mainly by Egress. @tnqn @xliuxu : do you think we can unify these two funcs: either both ignore or both include link local addresses.

Reply:
> I feel we can remove saveHostRoutes and change prepareOVSBridgeForK8sNode to use GetInterfaceConfig. Could you check?

It may not work for some corner cases if the uplink is configured by DHCP, and static routes exist on the host before attaching it to OVS. If the NIC is configured with DHCP, the routes on the host interface are configured from DHCP responses, then the static routes are possibly lost if we don't do it in the program.

Reply:
> Should we always ignore link local addresses? Could/should we change GetIPNetsByLink to do that too?

Having checked the code, it sounds good to me that ignoring the link-local addresses in `GetIPNetsByLink`.

Reply:
I went thought relevant functions today and agree they are kind of messy. I'm working on a patch trying to sort out these functions and moving business specific code out of util package.

Reply:
Yes, that is true. For this PR, do you agree to remove `GetAllIPNetsByName ` and change `GetIPNetsByLink`  to ignore link local addresses?

Reply:
`GetInterfaceConfig` is used by ExternalNode feature, not related to Egress.
Actually both `GetIPNetsByLink` and `GetAllIPNetsByName` are used to get the IPs that will be moved to OVS bridge port, one for K8s bridging mode, one for ExternalNode. Since they are essentially the same, I don't think there is a problem to make both ignore link local addresses.

Reply:
Ok. then let us change `GetIPNetsByLink` to ignore link local addresses. @aroradaman